### PR TITLE
[WAZO-1654] Update DOC guidelines

### PR DESCRIPTION
**Pull request by chartrandf**

### CODE REVIEW

File: content/uc-doc/contributors/contributing_doc.md [hunk 1]
```diff
@@ -2,140 +2,101 @@
...
+- Must be H2 or lower. _(H1 titles are automatically generated by React)_
+  ```
+  ## My Title 1
+
+  ### My Subtitle 1.1
+
+  ## My Title 2
+  ```
+
+### Anchors and Headings
```
Comment (fblackburn1) [inline content/uc-doc/contributors/contributing_doc.md:62]: wow nice I didn't see this new anchor style :tada:

File: content/uc-doc/contributors/contributing_doc.md [hunk 2]
```diff
@@ -2,140 +2,101 @@
...
+  ```
+- Must use `#` syntax to create a title
+- Must be H2 or lower. _(H1 titles are automatically generated by React)_
+  ```
+  ## My Title 1
+
+  ### My Subtitle 1.1
+
+  ## My Title 2
+  ```
```
Comment (fblackburn1) [inline content/uc-doc/contributors/contributing_doc.md:60]: Are there any reason to indent block by 2 spaces?

File: content/uc-doc/contributors/contributing_doc.md [hunk 3]
```diff
@@ -2,140 +2,101 @@
...
 -   There should be one and only one newline character at the end of
-    each file
+    each file.
 -   There should be no trailing whitespace at the end of lines
 -   Paragraphs must be wrapped and lines should be at most 100
-    characters long
-
-[^1]: `easy_install` can be found in the debian package
-    python-setuptools : `sudo apt-get install python-setuptools`
+    characters long
```
Comment (fblackburn1) [inline content/uc-doc/contributors/contributing_doc.md:102]: lol these lines are not wrapped to 100
I know the spec mention *at most* but it make a bit weird to have many limit when not apparent reason

File: content/uc-doc/contributors/contributing_doc.md [hunk 4]
```diff
@@ -2,140 +2,101 @@
...
+- Javascript
 
 ### Others {#others}
 
 -   There must be no warning nor error messages when building the
-    documentation with `make html`.
+    documentation with `yarn start`.
 -   There should be one and only one newline character at the end of
-    each file
+    each file.
```
Comment (fblackburn1) [inline content/uc-doc/contributors/contributing_doc.md:99]: Hehe you remove the newline char at the end of this file #fail :laughing:

File: content/uc-doc/contributors/contributing_doc.md [hunk 5]
```diff
@@ -2,140 +2,101 @@
...
+  ```
+  <!-- Correct -->
+  The Vitamins are in My Fresh California Raisins
+
+  <!-- Incorrect -->
+  The Vitamins Are In My Fresh California Raisins
+  ```
+- Must use `#` syntax to create a title
+- Must be H2 or lower. _(H1 titles are automatically generated by React)_
+  ```
```
Comment (fblackburn1) [inline content/uc-doc/contributors/contributing_doc.md:54]: You don't follow your own rule `Code blocks should have syntax highlight defined. Here's some values `

File: content/uc-doc/contributors/contributing_doc.md [hunk 6]
```diff
@@ -2,140 +2,101 @@
...
-       apt-get update
+    1. First item
+    2. Second item
+    3. ...
 
-Incorrect:
+### Code blocks
 
-    Bla bla bla:
+Code blocks should have syntax highlight defined. Here's some values :
```
Comment (fblackburn1) [inline content/uc-doc/contributors/contributing_doc.md:85]:
```suggestion
Code blocks should have syntax highlight defined. Here's some values:
```

File: content/uc-doc/contributors/contributing_doc.md (hunk 2 above)
Comment (chartrandf) [reply content/uc-doc/contributors/contributing_doc.md:60]: To keep the code block inside the list item.

File: content/uc-doc/contributors/contributing_doc.md (hunk 3 above)
Comment (chartrandf) [reply content/uc-doc/contributors/contributing_doc.md:102]: I just keep it from the previous guidelines. Honestly, I prefer to not break paragraphs and use **word ward/line wrap** inside my editor.

If you and @sduthil are alright with this, I'll remove it from the guideline.

Comment (sduthil) [reply content/uc-doc/contributors/contributing_doc.md:102]: I don't have opinion on this. With or without limit is fine with me, as long as we stick to the decision in all files.

File: content/uc-doc/contributors/contributing_doc.md (hunk 5 above)
Comment (chartrandf) [reply content/uc-doc/contributors/contributing_doc.md]: Solve, as discussed waiting on your feedback for «raw» content.

**Comment by chartrandf** [comment]
> I don't know if we should add a markdown linter
> Personally, I use a lot [markdownlint-cli](https://github.com/igorshubovych/markdownlint-cli) and for this file I have the following output (which are nice things to fix)

Gatsby comes with `prettier`. I added a new command `yarn format:uc-doc` and it fixe most of the issues.  I will create another PR to fix all other pages 😬.

1. It seems to fix all `markdownlint` issues except `line-length`. Based on your [answer](https://github.com/wazo-platform/wazo-platform.org/pull/187#discussion_r436808220). I'll check what we could do it we keep max length.
2. We could maybe add a test in Zuul for that?

File: content/uc-doc/contributors/contributing_doc.md (hunk 3 above)
Comment (fblackburn1) [reply content/uc-doc/contributors/contributing_doc.md]: I prefer to keep the limit
* some editor do not wrap line with default config (ex: vscode) and it's harder to read 
* some editor do not handle very well really long long line (when the wrapped text don't fit in the screen) (ex: vim)
* using CLI tool (e.g. grep) is more suitable/readable when you don't have a long line
* navigate through very long line can be painful if you don't use mouse (moving word by word)
* diff are generally painful with long line (I guess you can say the same thing when wrapping line and adding one word change all line length)
* One long line is harder to point of something (I rarely say, hey you made a typo line 1 column 25093)

File: package.json [hunk 7]
```diff
@@ -8,6 +8,7 @@
     "build": "./node_modules/.bin/gatsby build --prefix-paths",
     "develop": "./node_modules/.bin/gatsby develop",
     "format": "./node_modules/.bin/prettier --write src/**/*.{js,jsx} tests/*.js gatsby-node.js",
+    "format:uc-doc": "./node_modules/.bin/prettier --write content/uc-doc/*.md content/uc-doc/**/*.md",
```
Comment (fblackburn1) [inline package.json:11]: Not sure if that said that we cannot add file with more than 2 layers (without updating this file)? If so, it's pretty dangerous as configuration

**Comment by fblackburn1** [comment]
> > I don't know if we should add a markdown linter
> > Personally, I use a lot [markdownlint-cli](https://github.com/igorshubovych/markdownlint-cli) and for this file I have the following output (which are nice things to fix)
> 
> Gatsby comes with `prettier`. I added a new command `yarn format:uc-doc` and it fix most of the issues. I will create another PR to fix all other pages grimacing .
> 

Nice 

> 
>     2. We could maybe add a test in Zuul for that?

for sure, we can easily add it in the other PR

File: package.json (hunk 7 above)
Comment (chartrandf) [reply package.json]: Fixed an issue with the script... `globstar` was not working well when running with `yarn`. Yes, this script is checking on multiple level for markdown files.

> a globstar ** should match "zero or more directories and subdirectories searching for matches".

Can you explain why it's pretty dangerous?

File: content/uc-doc/contributors/contributing_doc.md (hunk 3 above)
Comment (chartrandf) [reply content/uc-doc/contributors/contributing_doc.md]: Alright, I added new rules in prettier to respect the 100 characters long rule. We will add in another PR a Zull test that will be sure our Markdown are valid.

**Pull request merged**